db/sql/00_msar.sql

```diff
@@ -185,35 +130,21 @@ $$ LANGUAGE plpgsql;
 
 
 CREATE OR REPLACE FUNCTION
-__msar.get_fully_qualified_object_name(sch_name text, obj_name text) RETURNS text AS $$/*
-Return the fully-qualified name for a given database object (e.g., table).
-
-Args:
-  sch_name: The schema of the object, quoted.
-  obj_name: The name of the object, unqualified and quoted.
-*/
-BEGIN
-  RETURN format('%s.%s', sch_name, obj_name);
-END;
-$$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
-
-
-CREATE OR REPLACE FUNCTION
-msar.get_fully_qualified_object_name(sch_name text, obj_name text) RETURNS text AS $$/*
+msar.build_qualified_name_sql(sch_name text, obj_name text) RETURNS text AS $$/*
```

Comment:
Since it returns a quoted value, shouldn't this be in the `__msar` namespace instead of `msar`?

Reply:
I don't have a strong opinion about this. I think `build_qualified_name_sql` could actually fit into `msar` or `__msar`. In 7f09220662bda25727dee32e3a9f643c3c54002c I've moved it to `__msar` though since you suggest that.

- I think it's okay in `msar` because the name of the function makes it clear that it's returning an SQL fragment, not a name. I think it's okay to return SQL fragments from functions as long as it's clear that it's a SQL fragment at not a name. In the case of this fully qualified name, there is no way to return it as a string without the quotes, so it has to be quoted. And in general our `msar` schema should allow for that, since eventually we won't have `__msar` anymore.

    That being said, I would like to avoid passing around fully-qualified names as single strings, just because it is so much _like_ a name that I think if falls into a kind of gray area that I'd like to avoid if possible.

- So with an eye towards removing the `build_qualified_name_sql` function eventually, I think it's fine to place it in `__msar` because that signals that it's deprecated.

I want to be clear: it's not that `__msar` is for "anything containing quoted stuff" and `msar` is for "nothing at all quoted ever". For example, in my [Record Summary PR](https://github.com/mathesar-foundation/mathesar/pull/3561/files), I wrote a function `msar.build_record_summary_query_from_template` which returns an entire SQL select query as a string. That string absolutely must contain quoted values. There's no way around it! But that's okay because there's no way to "unquote" that SQL. _(As a side note I should mention that when I authored that PR, my understanding of the purpose of `__msar` was that it was for private functions not called by the service layer, so I did define some functions within `__msar` which I'll now go back and move to `msar`.)_